docs/DeveloperGuide.adoc

```diff
@@ -319,6 +319,145 @@ image::UndoRedoActivityDiagram.png[width="650"]
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two different things.
 // end::undoredo[]
 
+// tag::recommend[]
+=== Recommending a workout
+==== Current implementation
+The user input is parsed into invoking a constructor based on the prefixes in the user input in `_RecommendCommandParser_`.There are only three acceptable prefixes. (i.e. `difficulty`, `calories`, `duration`) The instance created is then passed to `_RecommendCommand_` which uses `getFilteredWorkoutList()` and `getFilteredInternalList()` methods of `Model` to generate a random workout to recommend to the user.
+
+The sequence diagram of this entire process can be seen below:
+
+[[fig-RecommendCommandSD]]
+.Interactions inside `Logic` component for the *`recommend difficulty/beginner`* command.
+image::RecommendCommandSD.PNG[width="900"]
+
+==== Design Considerations
+
+==== Aspect: How recommend executes
+
+* **Alternative 1 (current choice):** Filters the workout list based on user's input and randomly generates a workout from that filtered list.
```

Comment:
Is the recommended workout "retrieved" randomly or "generated" randomly? Specify this to remove confusion.